buildSrc/subprojects/packaging/src/main/kotlin/org/gradle/gradlebuild/packaging/ShadedJarPlugin.kt

```diff
@@ -123,10 +123,11 @@ open class ShadedJarPlugin : Plugin<Project> {
         val configurationToShade = shadedJarExtension.shadedConfiguration
         val baseVersion: String by rootProject.extra
         val jar: TaskProvider<Jar> = tasks.withType(Jar::class).named("jar")
+        val jarInstallPath = findProperty("${project.name}ShadedJarInstallPath") ?: "shaded-jar/${base.archivesBaseName}-shaded-$baseVersion.jar"
```

Comment:
I think we should have a separate install task. Setting this property will break quite some tests and then you need to set/unset the property on the build.

Reply:
My intention was that this property should always be used with explicitly requested task `toolingApiShadedJar`, it should never be set when running tests depending on this task. 

In this way we can keep things simple, another install task is just adding some complexity. WDYT?

Reply:
I think we should have an `installToolingApiJar` task which does the installation, similar to the install task we have for installing a Gradle distribution. That separates the responsibilities for building the shaded jar and installing it somewhere else.